router/router.go

```diff
@@ -289,9 +289,9 @@ type ExpSegment struct {
 	Key string
 }
 
-// ParseExpSegment parses a regexp segment to ExpSegment.
-func ParseExpSegment(exp string) (*ExpSegment, error) {
-	if !strings.HasPrefix(exp, "{") || !strings.HasSuffix(exp, "}") {
+// parseExpSegment parses a regexp segment to ExpSegment.
+func parseExpSegment(exp string) (*ExpSegment, error) {
+	if !strings.HasSuffix(exp, "}") || !strings.HasPrefix(exp, "{") {
```

Comment:
why changing the order?

Reply:
The first `{` checking was done before the execution going into this func. So it was redundant.
In fact, we only need to check `}`.